src/mui/input/ReferenceInput.js

```diff
@@ -157,28 +160,37 @@ export class ReferenceInput extends Component {
             resource,
             label,
             source,
-            referenceRecord,
             allowEmpty,
             matchingReferences,
             basePath,
             onChange,
             children,
             meta,
+            translate,
         } = this.props;
-        if (!referenceRecord && !allowEmpty) {
-            return (
-                <Labeled
-                    label={
-                        typeof label === 'undefined' ? (
-                            `resources.${resource}.fields.${source}`
-                        ) : (
-                            label
-                        )
-                    }
-                    source={source}
-                    resource={resource}
-                />
-            );
+
+        const finalLabel =
+            typeof label === 'undefined'
+                ? `resources.${resource}.fields.${source}`
+                : label;
+        const translatedLabel = translate(finalLabel, { _: finalLabel });
+        const error =
+            matchingReferences && matchingReferences.error
+                ? translate(matchingReferences.error, {
+                      _: matchingReferences.error,
+                  })
+                : null;
+
+        if (React.Children.count(children) !== 1) {
+            throw new Error('<ReferenceInput> only accepts a single child');
+        }
+
+        if (matchingReferences === null) {
```

Comment:
I worry about another case:

1. The input displays and loads matching references => `matchingReferences` is null
2. The API returns an error => `matchingReferences` is `{ error: 'toto' }`
3. The user clicks "Refresh", which re-triggers the 2 fetch actions. But at that time, the value of `matchingReferences` is still an error, so the loader won't show...